src/compiler/checker.ts

```diff
@@ -12982,7 +12983,8 @@ namespace ts {
             const classLikeDeclaration = <ClassLikeDeclaration>container.parent;
             const classType = <InterfaceType>getDeclaredTypeOfSymbol(getSymbolOfNode(classLikeDeclaration));
             const baseClassType = classType && getBaseTypes(classType)[0];
-            if (!baseClassType) {
+
+            if (!(classType && classType.resolvedExtendedType)) {

```

Comment:
Instead of adding a field to `InterfaceType`, what if you just always did the `getClassExtendsClause` check below?